lib/src/analyzers/lint_analyzer/rules/rules_list/prefer_correct_edge_insets_constructor/visitor.dart

```diff
@@ -0,0 +1,220 @@
+      if (num.tryParse(argumentString) != 0) {
+        switch (index) {
+          case 0:
+            params.add('left: $argumentString');
+            break;
+          case 1:
+            params.add('top: $argumentString');
+            break;
+          case 2:
+            params.add('right: $argumentString');
+            break;
+          case 3:
+            params.add('bottom: $argumentString');
+            break;
+        }
+      }
+    }
+
+    _expressions[expression] = 'EdgeInsets.only(${params.join(', ')})';
+  }
+
+  void _validateSymmetric(InstanceCreationExpression expression) {
+    if (!expression.argumentList.arguments.every((element) =>
+        element.endToken.type == TokenType.INT ||
+        element.endToken.type == TokenType.DOUBLE)) {
+      return;
+    }
+
+    final params = <String, String>{};
+    for (final param in expression.argumentList.arguments) {
+      params[param.beginToken.toString()] = param.endToken.toString();
+    }
+
+    // EdgeInsets.symmetric(horizontal: 0, vertical: 12) -> EdgeInsets.symmetric(vertical: 12)
+    // EdgeInsets.symmetric(horizontal: 0, vertical: 0) -> EdgeInsets.zero
+    if (params.values.contains('0') || params.values.contains('0.0')) {
+      final arguments = <String>[];
+
+      for (final param in expression.argumentList.arguments) {
+        if (num.tryParse(param.endToken.toString()) != 0) {
+          arguments.add(
+            '${param.beginToken.toString()}: ${param.endToken.toString()}',
+          );
+        }
+      }
+
+      if (arguments.isEmpty) {
+        _expressions[expression] = 'EdgeInsets.zero';
+      } else {
+        _expressions[expression] =
+            'EdgeInsets.symmetric(${arguments.join(', ')})';
+      }
+
+      return;
+    }
+
+    // EdgeInsets.symmetric(horizontal: 12, vertical: 12) -> EdgeInsets.all(12)
+    if (params['horizontal'] == params['vertical']) {
+      _expressions[expression] = 'EdgeInsets.all(${params['horizontal']})';
+
+      return;
+    }
+  }
+
+  void _validateOnly(InstanceCreationExpression expression) {
+    if (!expression.argumentList.arguments.every((element) =>
+        element.endToken.type == TokenType.INT ||
+        element.endToken.type == TokenType.DOUBLE)) {
+      return;
+    }
+    final params = <String, String>{};
+    for (final param in expression.argumentList.arguments) {
+      params[param.beginToken.toString()] = param.endToken.toString();
+    }
+
+    if (params.isEmpty) {
+      return;
+    }
+    // EdgeInsets.only(left: 0, right: 12) -> EdgeInsets.only(right: 12)
+    if (params.values.contains('0') || params.values.contains('0.0')) {
+      final arg = <String>[];
+
+      for (final argument in expression.argumentList.arguments) {
+        if (num.tryParse(argument.endToken.toString()) != 0) {
+          arg.add(
+            '${argument.beginToken.toString()}: ${argument.endToken.toString()}',
+          );
+        }
+      }
```

Comment:
Looks like this block is duplicated several times.

Reply:
I am not found the same code block. they look alike, but different

Reply:
How so?


<img width="532" alt="Screenshot 2022-03-28 at 11 51 08" src="https://user-images.githubusercontent.com/21116742/160351678-7bc79628-475b-44b7-af21-9a3a778754ff.png">
<img width="608" alt="Screenshot 2022-03-28 at 11 50 59" src="https://user-images.githubusercontent.com/21116742/160351687-438c62e0-813b-46ae-bd97-2bd435f9ac67.png">